sdk/eventhubs/azure-messaging-eventhubs-checkpointstore-jedis/src/main/java/com/azure/messaging/eventhubs/checkpointstore/jedis/JedisRedisCheckpointStore.java

```diff
@@ -0,0 +1,162 @@
+import redis.clients.jedis.JedisPool;
+import redis.clients.jedis.Jedis;
+
+/**
+ * Implementation of {@link CheckpointStore} that uses Azure Redis Cache, specifically Jedis.
+ */
+public class JedisRedisCheckpointStore implements CheckpointStore {
+    private static final ClientLogger LOGGER = new ClientLogger(JedisRedisCheckpointStore.class);
+    private final JedisPool jedisPool;
+//    private final JacksonAdapter jacksonAdapter = new JacksonAdapter();
+    private static final JsonSerializer DEFAULT_SERIALIZER = JsonSerializerProviders.createInstance(true);
+    JedisRedisCheckpointStore(JedisPool jedisPool) {
+        this.jedisPool = jedisPool;
+
+    }
+
+    /**
+     * This method returns the list of partitions that were owned successfully.
+     *
+     * @param requestedPartitionOwnerships List of partition ownerships from the current instance
+     * @return Null
+     */
+    @Override
+    public Flux<PartitionOwnership> claimOwnership(List<PartitionOwnership> requestedPartitionOwnerships) {
+        return null;
+    }
+
+    /**
+     * This method returns the list of checkpoints from the underlying data store, and if no checkpoints are available, then it returns empty results.
+     *
+     * @param fullyQualifiedNamespace The fully qualified namespace of the current instance of Event Hub
+     * @param eventHubName The Event Hub name from which checkpoint information is acquired
+     * @param consumerGroup The consumer group name associated with the checkpoint
+     * @return Null
+     */
+    @Override
+    public Flux<Checkpoint> listCheckpoints(String fullyQualifiedNamespace, String eventHubName, String consumerGroup) {
+        String prefix = prefixBuilder(fullyQualifiedNamespace, eventHubName, consumerGroup);
+        ObjectSerializer serializer = DEFAULT_SERIALIZER;
+
+        try (Jedis jedis = jedisPool.getResource()) {
+            Set<String> members = jedis.smembers(prefix);
+            if (members.isEmpty()) {
+                return Flux.error(new IllegalArgumentException()); // no checkpoints to list
+            }
+            ArrayList<Checkpoint> list = new ArrayList<>();
+            for (String member : members) {
+                //get the associated JSON representation for each for the members
+                List<String> checkpointJsonList = jedis.hmget(member, "checkpoint");
+                String checkpointJson;
+                if (checkpointJsonList.size() == 0) {
+                    return Flux.error(new NoSuchElementException());
+                }
+                else {
+                    checkpointJson = checkpointJsonList.get(0);
+                }
+                Checkpoint checkpoint = serializer.deserializeFromBytes(checkpointJson.getBytes(StandardCharsets.UTF_8), TypeReference.createInstance(Checkpoint.class));
+                list.add(checkpoint);
+            }
+            jedisPool.returnResource(jedis);
+            return Flux.fromStream(list.stream());
```

Comment:
You can use Flux.fromIterable rather than wrapping it in a stream again.  Creating a list just to temporarily hold these results seems like a lot of overhead... is it possible to use Flux.create?

Reply:
That's true. I've gone ahead and changed it such that all the Flux.fromStream are Flux.fromIterable instead.